nubia/internal/interactive.py

```diff
@@ -138,25 +139,26 @@ def evaluate_command(self, cmd, args, raw):
                 # not implemented error code
                 return 99
 
-    def run(self):
+    async def run(self):
         prompt = self._build_cli()
         self._status_bar.start()
         try:
             while True:
                 try:
-                    text = prompt.prompt(
-                        PygmentsTokens(self._get_prompt_tokens()),
-                        rprompt=PygmentsTokens(self._status_bar.get_rprompt_tokens()),
-                    )
+                    with patch_stdout():
```

Comment:
> This implements a context manager that ensures that print statements within it won’t destroy the user interface. The context manager will replace sys.stdout by something that draws the output above the current prompt, rather than overwriting the UI.

I'm not entirely familiar with promp toolkit. But what would be the reason for adding it here together with the asyncio change?

Reply:
My turn to apologise for not getting back to you sooner. (notifications issues)

Please see the detailed explanation in [this part of the `prompt_toolkit` doc](https://python-prompt-toolkit.readthedocs.io/en/master/pages/asking_for_input.html?highlight=patch_stdout#prompt-in-an-asyncio-application).